### PR TITLE
Add most read page and refactor container

### LINKED ISSUE #4472: Parent Issue: Most Read as a user-facing URL
Currently all WS sites have a Most Read page as a URL users can visit. The URL is of the format:
`bbc.com/<service>/popular/read` eg. /igbo/popular/read

If we want to maintain this url, and I think at present we do, then we will need to recreate this page somewhere, for all services, and it seems making use of the new most read component in a page is logical.

**Describe the solution you'd like**
I would like a page in Simorgh to be available for each service of this format:
`bbc.com/<service>/popular/read`
If a user visits that page I would like them to be able to see a list of popular news stories from that service.
Accessibility - Most read should be within the main container and the h1 should be visually displayed as ‘Most read’ and the skip to content link should anchor here (note there should not be a h2 for most read or any other regions for most read within the main region) 

The outcome of this will be that we have a new page type and route in Simorgh, which will initially be available locally, and we can then get feedback from UX and make changes accordingly.

**Describe alternatives you've considered**
In theory, in a world in which we will be building other most popular lists of content there could be an argument that using the url `bbc.com/<service>/popular` might be better, since it would allow us to stack Most Read, Most Watched, Most popular topics etc in a single page.
But that would then require a bunch of redirects from /popular/read to /popular

**Testing notes**
To be tested according to design doc across various breakpoints.Also the ATI and chartbeat testing for this new page.

- Unit tests
- Integration tests
- One cypress test to cover this as it is a new page type.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.
**Additional context**
This piece of work is potentially a thing we could do, or not do. Before we get deep into it I'd be interested in an estimate of how much work is needed to make this happen for all sites.

### PR DESCRIPTION
**Pull request by thekp**
Resolves #4472 

Dependant on this PR: https://github.com/bbc/simorgh/pull/6171 (updating getInitialData)

**Overall change:**
_Create most read page that is using server side fetched data. Refactored most read to accept a wrapper prop so that each page pass it's own wrapper._

Local Example: 
image

To see locally:

 http://localhost:7080/mundo/23263889
http://localhost:7080/pidgin/articles/cwl08rd38l6o
http://localhost:7080/pidgin/popular/read
http://localhost:7080/pidgin

**Code changes:**

- Setup route for most read page, that uses service from getInitialData to generate the correct path to fetch data from (`/$service/mostread.json`).
- Added most read page, we used the h1 specs specified by UX.
- Refactored most read, so it accepts a wrapper prop, which allows us to pass a wrapper for most read on each page type. For example on front pages we pass in a wrapper that contains a section label and desired wrapper for the front page most read requirements.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by thekp** [comment]
Cypress tests will be added a part of this PR: https://github.com/bbc/simorgh/issues/6169
A11y: Will be done once this page is available of TEST.

**Comment by thekp** [comment]
We have plans to consolidate the most read container with radio schedules container: https://github.com/bbc/simorgh/issues/6219

**Comment by thekp** [comment]
Whilst reviewing/testing this. I realised this PR doesn't take variants into account

Edit: I've resovled this problem by passing variant into getIntialData and adding a getMostReadUrls function

File: src/app/containers/MostRead/Canonical/index.jsx
Comment (DenisHdz) [inline src/app/containers/MostRead/Canonical/index.jsx:69]: Shall we log the error here as well, as we've done with Radio Schedules? https://github.com/bbc/simorgh/pull/6142/files#diff-91ab1611f147d3d33e441ceb72157ed3R143

Comment (thekp) [reply src/app/containers/MostRead/Canonical/index.jsx:69]: yeah definitely, I wanted to hold out doing it in this PR as I wanted to make it the same as the radio schedules container so I'll wait till that is merged.

We have this issue: https://github.com/bbc/simorgh/issues/6219 to consolidate the two containers after the work has been merged.

File: src/app/containers/MostRead/Canonical/index.test.jsx
Comment (DenisHdz) [inline src/app/containers/MostRead/Canonical/index.test.jsx]: We can avoid this test since the following one is assuming that there also X number of `li`s

Comment (thekp) [reply src/app/containers/MostRead/Canonical/index.test.jsx]: yes good spot 👍

File: src/app/containers/ChartbeatAnalytics/utils/index.js
Comment (Bopchy) [inline src/app/containers/ChartbeatAnalytics/utils/index.js:38]: Should we add a unit test for this?

File: src/app/lib/utilities/getMostReadUrls/index.test.js
Comment (Bopchy) [inline src/app/lib/utilities/getMostReadUrls/index.test.js]:
```suggestion
describe('getMostReadEndpoint', () => {
```

File: src/app/lib/utilities/getMostReadUrls/index.test.js
Comment (Bopchy) [inline src/app/lib/utilities/getMostReadUrls/index.test.js]:
```suggestion
describe('getLocalMostReadEndpoint', () => {
```

File: src/app/lib/utilities/getMostReadUrls/index.test.js
Comment (Bopchy) [inline src/app/lib/utilities/getMostReadUrls/index.test.js]:
```suggestion
  it('should return endpoint when passed service & variant', () => {
```

File: src/app/pages/MostReadPage/index.jsx
Comment (Bopchy) [inline src/app/pages/MostReadPage/index.jsx]: I think `HeadingOne` sounds a bit too generic. Maybe `MostReadHeader` or `PageHeader`? 🤷🏽‍♀️

File: src/app/containers/App/App.test.jsx
Comment (Bopchy) [inline src/app/containers/App/App.test.jsx:190]: Is it worth changing the variant to `default`, or the service to one that has a `simp` variant, so that this matches the actual data that we have?

**Review by Bopchy — COMMENTED** [review]
Looks really great 👍! Left a couple of comments.

File: src/app/containers/App/App.test.jsx
Comment (thekp) [reply src/app/containers/App/App.test.jsx:190]: https://github.com/bbc/simorgh/blob/61361b85439cbaa8c11107a6552cfb998ade7e88/src/app/containers/App/App.test.jsx

so it's expecting simp because we have added it here to have variant of `/simp` 🤦. What I'm going to do is change it to a service with a variant so the tests make more sense.

File: src/app/pages/ArticlePage/index.jsx
Comment (DenisHdz) [inline src/app/pages/ArticlePage/index.jsx]: Should we call this constant `ArticleMostReadSection` as we do with the Front Page https://github.com/bbc/simorgh/pull/6175/files ?

Comment (thekp) [reply src/app/pages/ArticlePage/index.jsx]: yes, that makes more sense

File: src/app/lib/utilities/getMostReadUrls/index.js
Comment (DenisHdz) [inline src/app/lib/utilities/getMostReadUrls/index.js:1]: This file looks great, just wondering if we could have just one function and depending on the environment `process.env.SIMORGH_APP_ENV` then return the right path.

File: src/app/pages/MostReadPage/index.jsx
Comment (DenisHdz) [inline src/app/pages/MostReadPage/index.jsx]: Do you need this?

Comment (thekp) [reply src/app/pages/MostReadPage/index.jsx]: Just tested it locally without this flex-grow and it still works as expected 👍 Will remove now

File: src/app/lib/utilities/getMostReadUrls/index.js
Comment (thekp) [reply src/app/lib/utilities/getMostReadUrls/index.js:1]: the getLocalMostReadEndpoint is for when we are on storybook and it would be difficult to differentiate storybook environment from our others.

Comment (DenisHdz) [reply src/app/lib/utilities/getMostReadUrls/index.js:1]: Fair enough 👍

**Review by DenisHdz — APPROVED** [review]
Thanks for the changes. The three types look ace 💯

**Aricle Page**
MostRead-ArticlePage

**Front Page**
MostRead-FrontPage

**Most Read Page**
MostRead-Page

**Review by Bopchy — APPROVED** [review]
Thanks for making the changes.

**Comment by PriyaKR** [comment]
Padding seems to be different from the designs.Refer screenshot.
Screenshot 2020-04-24 at 16 34 16

**Comment by thekp** [comment]
Hi @PriyaKR, 
We add some extra spacing here around the most read container itself here, because it's needed for other page types, it's a margin of 24px.
image

so in total we have  a padding bottom of 8px on our h1 here and 24px on the container below to get a total of 32

image

**Comment by thekp** [comment]
UX review: ✅ 


I have had a UX review for the changes on this PR

Notes from UX review: 

UX confirmed that the spacing around the h1 should be the tablet viewport should be the same as the desktop viewport.

**Comment by PriyaKR** [comment]
Looks good to me.

**Comment by greenc05** [comment]
Do we have an issue so we remember to carry out the a11y swarm on test @thekp

**Comment by thekp** [comment]
@greenc05
https://github.com/bbc/simorgh/issues/6185

Yep we do. I'll move it into progress once I get this page on to TEST env.

**Comment by thekp** [comment]
image
I just noticed on the chromatic snapshots that I need to add section label to CPS pages

**Review by Bopchy — APPROVED** [review]

**Review by DenisHdz — APPROVED** [review]
Looks good on Story Pages too

stys

**Pull request merged**
